### PR TITLE
Drag&Drop fix

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

During implementation TreeView for `LibrarySearchView` Drag&Drop functionality was lost. I returned it back, but then I thought, that functionality in LibraryView and LibrarySearchView is in fact the same.
So, I decided to move it in the separate class-helper. And moved this class in Utilities folder.

Class, that helps drag&drop is called `LibraryDragAndDrop`. It's used is both: LibraryView and LibrarySearchView.

### Declarations

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@Benglin

### CODE REVIEW

File: src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        /// <summary>
+        /// Position of mouse, when user clicks on button.
+        /// </summary>
+        public Point StartPosition;
+
+        /// <summary>
+        /// Indicates whether item is dragging or not, so that there won't be more than one DoDragDrop event.
+        /// </summary>
+        private bool IsDragging;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs]: Data member should be named `isDragging`.

File: src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System.Windows;
+
+namespace Dynamo.Wpf.Utilities
+{
+    public class LibraryDragAndDrop
+    {
+        /// <summary>
+        /// Position of mouse, when user clicks on button.
+        /// </summary>
+        public Point StartPosition;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs]: Should be named `startPosition` and made `private readonly`. Then the only way to initialize this is through a constructor, does that make sense?

File: src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        public void MouseMove(FrameworkElement sender, Point currentPosition, NodeSearchElementViewModel node)
+        {
+            // If item was dragged enough, then fire DoDragDrop. 
+            // Otherwise it means user click on item and there is no need to fire DoDragDrop.
+            if ((System.Math.Abs(currentPosition.X - StartPosition.X) > SystemParameters.MinimumHorizontalDragDistance ||
+                System.Math.Abs(currentPosition.Y - StartPosition.Y) > SystemParameters.MinimumVerticalDragDistance) &&
+                !IsDragging)
+            {
+                StartDrag(sender, node);
+            }
```
Comment (Benglin) [inline src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs]: For readability, I'd recommend this:

```csharp
public void MouseMove(...)
{
    if (isDragging)
        return;

    var deltaX = System.Math.Abs(currentPosition.X - StartPosition.X);
    if (deltaX < SystemParameters.MinimumHorizontalDragDistance)
        return;

    var deltaY = ...;

    StartDrag(...);
}
```

File: src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs [hunk 4]
```diff
@@ -20,6 +20,7 @@ public partial class LibraryView : UserControl
     {
         // See OnExpanderButtonMouseLeftButtonUp for details.
         private bool ignoreMouseEnter;
+        private LibraryDragAndDrop drag_drop = new LibraryDragAndDrop();
```
Comment (Benglin) [inline src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs]: Maybe call this `dragDropHelper`?

**Comment by Benglin** [comment]
Hi @aosyatnik, thanks for consolidating the duplicated code, well done! Sorry for being late on this, some changes and we'll be good to go!

**Comment by aosyatnik** [comment]
Hi, @Benglin , I've updated my code. Please, take a look. Thanks!

**Comment by pboyer** [comment]
@Benglin Any updates here?

**Comment by Benglin** [comment]
Once again, sorry guys. This looks great!

**Pull request merged**